src/plugins/core/sheet.ts

```diff
@@ -550,42 +541,34 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
     return sheet;
   }
 
-  private moveSheet(sheetId: UID, direction: "left" | "right") {
+  private moveSheet(sheetId: UID, delta: number) {
     const orderedSheetIds = this.orderedSheetIds.slice();
     const currentIndex = orderedSheetIds.findIndex((id) => id === sheetId);
     const sheet = orderedSheetIds.splice(currentIndex, 1);
-    let index =
-      direction === "left"
-        ? this.findIndexOfPreviousVisibleSheet(currentIndex - 1, orderedSheetIds)
-        : this.findIndexOfNextVisibleSheet(currentIndex + 1, orderedSheetIds);
-    if (index === undefined) {
-      index = orderedSheetIds.length;
-    }
+    let index = this.findIndexOfTargetSheet(currentIndex, delta);
     orderedSheetIds.splice(index, 0, sheet[0]);
     this.history.update("orderedSheetIds", orderedSheetIds);
   }
 
-  private findIndexOfPreviousVisibleSheet(current: HeaderIndex, orderedSheetIds: UID[]) {
-    while (current >= 0 && !this.isSheetVisible(orderedSheetIds[current])) {
-      current--;
-    }
-    if (current === -1) {
-      throw new Error("There is no previous visible sheet");
-    }
-    return current;
-  }
-
-  private findIndexOfNextVisibleSheet(current: HeaderIndex, orderedSheetIds: UID[]) {
-    while (current < orderedSheetIds.length && !this.isSheetVisible(orderedSheetIds[current])) {
-      current++;
+  private findIndexOfTargetSheet(currentIndex: HeaderIndex, deltaIndex: number): number {
+    while (deltaIndex != 0 && 0 <= currentIndex && currentIndex <= this.orderedSheetIds.length) {
+      if (deltaIndex > 0) {
+        currentIndex++;
+        if (this.isSheetVisible(this.orderedSheetIds[currentIndex])) {
+          deltaIndex--;
+        }
+      } else if (deltaIndex < 0) {
+        currentIndex--;
+        if (this.isSheetVisible(this.orderedSheetIds[currentIndex])) {
+          deltaIndex++;
+        }
+      }
     }
-    if (
-      current === orderedSheetIds.length - 1 &&
-      !this.isSheetVisible(orderedSheetIds[current - 1])
-    ) {
-      return undefined;
+    if (deltaIndex === 0) {
+      return currentIndex;
+    } else {
+      throw new Error("There is not enough visible sheets");
```

Comment:
Open question, do we consider that using `allowDispatch` ensure us that this message will never be thrown to the user ? Or should we translate it, just in case ? Do not really know.

Reply:
I'd say this shouldn't happen so we don't care, but I don't really mind either way

Reply:
@rrahir @LucasLefevre Any through about this ?

Reply:
Does not cost much to translate ¯\_(ツ)_/¯